Client/Frontend/Library/BookmarkDetailPanel.swift

```diff
@@ -180,14 +181,18 @@ class BookmarkDetailPanel: SiteTableViewController {
         }
     }
 
+    func updateSaveButton(text: String?){
+        navigationItem.rightBarButtonItem?.isEnabled = bookmarkNodeType != .bookmark || (text != nil ? URIFixup.getURL(text!)?.host != nil : false)
+    }
+
     func save() -> Success {
         if isNew {
             if bookmarkNodeType == .bookmark {
-                guard let bookmarkItemURL = self.bookmarkItemURL else {
+                guard let bookmarkItemURL = self.bookmarkItemURL, let fixedUpBookmarkItemURL = URIFixup.getURL(bookmarkItemURL) else {
```

Comment:
If you're validating already to disable the save button when the URL field changes, you don't need to do anything here.

Reply:
The point here isn't to validate; it's to (if needed) change the input to be acceptable to the backend. For example, it will change "google.com" (which the backend won't accept) to "http://google.com/".

Reply:
Yes, but I believe we would rather just prevent the user from saving if they don't enter a valid URL. Using the `URIFixup` I believe will automatically prepend an `http://` and we would rather not do something that would cause users to create non-HTTPS bookmarks. This is why I would rather pre-fill the input with `https://` and let the user provide the rest.

Reply:
This complicates matters, since if we can't use URIFixup here we can't use it to validate either, as there's no guarantee the user doesn't delete the pre-filled "https://", resulting in an invalid input which passes validation.

I can think of 3 approaches here:

1. Modify URIFixup to use https instead of http. This seems the best unless it would cause problems elsewhere.
2. Modify URIFixup to optionally allow the preferred protocol to be specified.
3. Use custom validation code instead of using URIFixup.

Which approach would you prefer?

Reply:
It is perfectly fine to use `URIFixup` to validate the URL. But, we want to use the opportunity to steer users towards including `https://` by pre-filling the field. We still want to allow users to back out the `https://` and manually enter `http://` themselves (not all websites support HTTPS). What we are explicitly trying to avoid is letting users just enter something like `google.com` and having it automatically "correct" it to a non-HTTPS URL without their explicit input. If you just follow the comments I've already posted to your PR, you'll arrive at an acceptable solution (only validate for purposes of enabling/disabling the Save button but don't try to correct the user's input).

Reply:
Using URIFixup just to validate won't avoid the original bug when the user deletes the pre-filled text.

Therefore, I have rewritten the validation to just check for a valid scheme/host and removed the correction of the user's input.

Reply:
Sorry, my bad. I was under the assumption that it would not validate if it was missing the scheme+host. As long as we disable the "Save" button when scheme+host is missing, that would be good enough to fix this. Keep the validation code in BookmarkDetailPanel.swift though.

Reply:
The latest revision does that. Do you see any remaining issues?

Reply:
That looks good. The only additional request I have is that you remove the `text:` argument from `updateSaveButton()` and simply use the value of `bookmarkItemURL` directly.

Reply:
Done.